Source/Plugins/Core/com.equella.core/src/com/tle/core/connectors/blackboard/service/impl/BlackboardRESTConnectorServiceImpl.java

```diff
@@ -606,6 +596,56 @@ public boolean supportsReverseSort() {
     }
   }
 
+  private Token getUserSessionAuth() {
+    Optional<Token> auth =
+        userSessionService.getAttribute(BlackboardRESTConnectorConstants.USER_SESSION_AUTH_KEY);
+    if (!auth.isPresent()) {
+      throw new AuthenticationException(
+          "User authentication details are not available in the session");
+    }
+    return auth.get();
```

Comment:
Should be an orElseThrow:

```suggestion
    return userSessionService.getAttribute(BlackboardRESTConnectorConstants.USER_SESSION_AUTH_KEY)
      .orElseThrow(
        () -> new AuthenticationException("User authentication details are not available in the session"));
```